From 789187871b907617eaa58f8b097ae7f011274de4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michael=20A=C3=AFv=C3=A1zis?= <michael.aivazis@para-sim.com>
Date: Fri, 11 Aug 2023 14:13:14 -0700
Subject: [PATCH] cmake: added the missing files to the {pyre::memory} bindings

---
 .cmake/pyre_pyre.cmake | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/.cmake/pyre_pyre.cmake b/.cmake/pyre_pyre.cmake
index 24b3fe581..97593d922 100644
--- a/.cmake/pyre_pyre.cmake
+++ b/.cmake/pyre_pyre.cmake
@@ -139,6 +139,7 @@ function(pyre_pyreModule)
     extensions/pyre/grid/packings.cc
     extensions/pyre/grid/shapes.cc
     extensions/pyre/memory/__init__.cc
+    extensions/pyre/memory/heaps.cc
     extensions/pyre/memory/maps.cc
     extensions/pyre/timers/__init__.cc
     extensions/pyre/timers/process_timers.cc
