diff --git a/lib/pyre/timers/ProcessClock.icc b/lib/pyre/timers/ProcessClock.icc
index 1c80fcef8..b654c4dc2 100644
--- a/lib/pyre/timers/ProcessClock.icc
+++ b/lib/pyre/timers/ProcessClock.icc
@@ -26,7 +26,7 @@ pyre::timers::ProcessClock::now() -> time_point_type
 // otherwise
 #else
     // complain
-    throw std::runtime_error("this platform doesn't have support for CLOCK_PROCESS_CPUTIME_ID")
+    throw std::runtime_error("this platform doesn't have support for CLOCK_PROCESS_CPUTIME_ID");
 #endif
 }
 
